alternatives.c

```diff
@@ -1290,6 +1290,32 @@ static int listServices(const char *altDir, const char *stateDir, int flags) {
     return 0;
 }
 
+int dirExists(const char *path) {
+    struct stat stats;
+    stat(path, &stats);
+
+    if (S_ISDIR(stats.st_mode))
```

Comment:
I believe that at this point we'll be reading potentially uninitialized memory; I don't believe the `stat` system call (or C library wrapper) will necessarily initialize the target structure on error.

So we definitely want to bail early if we get an error.

Reply:
Thanks for catching!